docs/docs/deployment/docker.md

```diff
@@ -0,0 +1,109 @@
+---
+sidebar_position: 5
+---
+
+# Docker
+
+Many cloud providers support running Docker containers. Thus, Exograph can be deployed through Docker. This document provides instructions on performing specific tasks and running an Exograph server locally for development using Docker.
+
+:::info Explicitly supported providers
+Exograph explicitly supports running a server in [Fly.io](../flyio) and [Railway](../railway). While the underlying mechanism uses Docker, you should follow the specific deployment guides for these providers.
+
+Exograph does **not** use a Docker container for AWS Lambda or Cloudflare Workers. To deploy in those environments, please follow the [AWS Lambda](../aws-lambda) or [Cloudflare Workers](../cloudflare-workers) guides.
+:::
+
+Exograph provides three Docker images:
+
+- `ghcr.io/exograph/cli`: It includes only the [`exo` CLI](../cli-reference/development) and may be used for development tasks such as building the `exo_ir` file and computing the database schema. Note that you cannot run `exo yolo` or `exo dev` using this image (you can use `ghcr.io/exograph/dev` for this purpose).
+- `ghcr.io/exograph/server`: It includes only the Exograph server and forms the basis for running it in production.
+- `ghcr.io/exograph/dev`: It includes the Exograph server, the `exo` CLI, and Postgres 16. Since it is optimized for development, it is a rather large image, and it is **not recommended** for production use. Typically, you will enter [into a shell for this image to perform development tasks, including `exo yolo` and `exo dev`](../getting-started/docker).
+
+Let's see how you can use these images for specific tasks.
+
+## Building the `exo_ir` file
+
+To run an Exograph server, you must build an `exo_ir` file. While you would typically build this file using the [build](../cli-reference/development/build.md) command to use the locally installed `exo` CLI, you can also build this file using a Docker container using the `ghcr.io/exograph/cli` image (you may also use `ghcr.io/exograph/dev`).
+
+```sh
+# shell-command-next-line
+docker run --rm --platform linux/x86_64 \
+  --mount type=bind,source="$(pwd)",target=/usr/src/app \
+  -it ghcr.io/exograph/cli:latest bash -c "exo build"
+```
+
+This will build the `exo_ir` file and put it in the `target` directory.
+
+## Running the server
+
+To run the server locally, you will need to pass the Postgres connection URL such that the Docker container can resolve it.
+
+```sh
+# shell-command-next-line
+docker run --rm --platform linux/x86_64 \
+  --mount type=bind,source="$(pwd)",target=/usr/src/app \
+  -e EXO_POSTGRES_URL=<postgres-url> \
+  -e EXO_INTROSPECTION=true \
```

Comment:
is `EXO_JWT_SECRET` missing here?

Reply:
Good suggestion. I added it (and `EXO_OIDC_URL`) to the list of additional env to pass.